extensions/panache/mongodb-panache/runtime/src/main/java/io/quarkus/mongodb/panache/runtime/MongoOperations.java

```diff
@@ -265,7 +267,7 @@ private static MongoCollection mongoCollection(Object entity) {
     }
 
     private static MongoDatabase mongoDatabase(MongoEntity entity) {
-        MongoClient mongoClient = Arc.container().instance(MongoClient.class).get();
+        MongoClient mongoClient = Arc.container().instance(MongoClient.class, Default.Literal.INSTANCE).get();
```

Comment:
do not worked sadly. I need help here. I do not understand why panache test failed. mongoClient return by arc is empty. How debug that ?
I have some code I try to execute but without success:
LOGGER.debug("DUFF DEBUG:");
        LOGGER.debug(mongoClient);
        Set<Bean<?>> beans = Arc.container().beanManager().getBeans(MongoClient.class);
        for (Bean<?> bean : beans) {
            if (bean.getName() == null) {
                MongoClient mc1 = Arc.container().instance(MongoClient.class).get();
                LOGGER.debug("DUFF DEBUG mc1:");
                LOGGER.debug(mc1);
            } else {
                MongoClient mc2 = (MongoClient) Arc.container().instance(bean.getName()).get();
                LOGGER.debug("DUFF DEBUG mc2:");
                LOGGER.debug(mc2);
            }
        }
        InstanceHandle<? extends AbstractMongoClientProducer>mongoClientProducerHandle = Arc.container().instance(AbstractMongoClientProducer.class);
        if (!mongoClientProducerHandle.isAvailable()) {
            throw new IllegalStateException("No mongo client producer has been defined.");
        }
        InstanceHandle<MongoClient>mongoClientHandle = Arc.container().instance(MongoClient.class);
        if (!mongoClientHandle.isAvailable()) {
            throw new IllegalStateException("No connection has been defined.");
        }
I do not know where log is stored...

Reply:
I can have a look at your PR later if you don't find the issue, will need to checkout your branch.
As I can see in the code, you need to access the instance of the bean with the qualifier `<default>` as it's the one you use in the recorder for the default client.

So the code should look like 
```
MongoClient mongoClient = Arc.container().instance(MongoClient.class, "<default>").get();
```

Or you can rollback your change and document the fact that Mongodb with Panache didn't works yet with multiple client. Because the previous code `MongoClient mongoClient = Arc.container().instance(MongoClient.class).get();` should works if there is only one bean in the beanFactory (if it didn't work it means that the bean is removed so maybe  you need to add them as unremovable beans)

Reply:
I do not succeed ;-(

Reply:
@dufoli this is very strange, as it gets the bean from Arc only at runtime it should be there. 

Can you try to annotate the generated bean with `io.quarkus.arc.Unremovable` to be 100% sure that Arc didn't remove the bean? You should add the annotation during the creation of the MongoClientProducerBean in `MongoClientProcessor.createMongoClientProducerBean`

Reply:
I end with the same understanding, it is removed by optimization. (Default is added on arc so no need to add it.)
So I try to add it at few place but bean as not bean generated so I add it in the listener created function but still not work... strange